Automattic/wp-calypso
# FSE: Disable Block Directory on Simple sites

#### Changes proposed in this Pull Request

Disable Block Directory via FSE's Block Inserter Modifications module by unregistering the `block-directory` plugin and registering our own to render `no results` state.

⚠️ This implementation disables Block Inserter on Simple sites **only** (`wp-admin` + `iframed` editors).
⚠️ This [module](https://github.com/Automattic/wp-calypso/tree/e9e41bdda49da4e93f16580f9fdb1acdcb1de240/apps/full-site-editing/full-site-editing-plugin/block-inserter-modifications/disable-block-directory) should be removed once we start supporting Block Directory.

#### Testing instructions

- Enable edge sticker on your Simple test site,
- Open that site sandboxed and verify that it's using Gutenberg v8.5.1,
- Run `yarn dev --sync` to upload local FSE `dev` build to your sandbox,
- Once FSE has uploaded, refresh the page and open the Block Inserter,
- Try searching for something that should return no results - you should see the `No results found.` message,
- Try searching for `css` - you should see the no results message **and** the Contextual Tip about our Customizer,
- Build and upload the plugin to your AT site and make sure the Block Directory isn't disabled there.

#### Screenshots

| Before<br>(no results)  | Before<br>(no res. + cont. tip) | After<br>(no results) | After<br>(no res. + cont. tip) |
| ------------- | ------------- | ------------- |  ------------- |
| ![image](https://user-images.githubusercontent.com/1451471/87157177-221b2300-c2be-11ea-90e6-6e9eb7dcd700.png) | ![image](https://user-images.githubusercontent.com/1451471/87157200-2b0bf480-c2be-11ea-97d5-28404e255bdc.png) | ![image](https://user-images.githubusercontent.com/1451471/87294348-eb364e80-c503-11ea-89dc-28460a6cc104.png) | ![image](https://user-images.githubusercontent.com/1451471/87294363-f1c4c600-c503-11ea-92b9-0285be0eaeb1.png) |

Fixes #43677
